TrafficCapture/trafficCaptureProxyServer/src/main/java/org/opensearch/migrations/trafficcapture/proxyserver/netty/HeaderAdderHandler.java

```diff
@@ -0,0 +1,52 @@
+package org.opensearch.migrations.trafficcapture.proxyserver.netty;
+
+import java.nio.charset.StandardCharsets;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandlerAdapter;
+
+public class HeaderAdderHandler extends ChannelInboundHandlerAdapter {
+    private static final ByteBuf CRLF_BYTE_BUF =
+        Unpooled.unreleasableBuffer(Unpooled.wrappedBuffer("\r\n".getBytes(StandardCharsets.UTF_8)));
+    private static final ByteBuf LF_BYTE_BUF =
+        Unpooled.unreleasableBuffer(Unpooled.wrappedBuffer("\n".getBytes(StandardCharsets.UTF_8)));
+    boolean insertedHeader = false;
+    private final ByteBuf headerLineToAdd;
+    boolean useCarriageReturn;
+
+    public HeaderAdderHandler(ByteBuf headerLineToAdd) {
+        this.headerLineToAdd = headerLineToAdd.retain();
```

Comment:
This feels strange to have retain here without a release.

Reply:
Yeah - that's a bug.  I'll add the release to the unregistered.  It didn't come up as an issue because the one that is sent in is a constant unreleasable buffer,.